torchtext/experimental/vectors.py

```diff
@@ -83,30 +92,38 @@ def glo_ve(name="840B", unk_tensor=None, root='.data'):
             - twitter.27B
             - 6B
         unk_tensor (Tensor): a 1d tensor representing the vector associated with an unknown token.
-        root: folder used to store downloaded files in (.data)
-
+        root (str): folder used to store downloaded files in (.data)
+        validate_file (bool): flag to determine whether to validate the downloaded files checksum.
+                              Should be `False` when running tests with a local asset.
     Returns:
         Vectors: a Vectors object.
 
     """
     urls = {
-        '42B': 'http://nlp.stanford.edu/data/glove.42B.300d.zip',
-        '840B': 'http://nlp.stanford.edu/data/glove.840B.300d.zip',
-        'twitter.27B': 'http://nlp.stanford.edu/data/glove.twitter.27B.zip',
-        '6B': 'http://nlp.stanford.edu/data/glove.6B.zip',
+        '42B': 'https://nlp.stanford.edu/data/glove.42B.300d.zip',
+        '840B': 'https://nlp.stanford.edu/data/glove.840B.300d.zip',
+        'twitter.27B': 'https://nlp.stanford.edu/data/glove.twitter.27B.zip',
+        '6B': 'https://nlp.stanford.edu/data/glove.6B.zip',
     }
 
     url = urls[name]
-    vectors_file_path = os.path.join(root, os.path.splitext(os.path.basename(url))[0] + '.pt')
-    if os.path.isfile(vectors_file_path):
-        return(torch.load(vectors_file_path))
-
-    downloaded_file_path = download_from_url(url, root=root)
+    cached_vectors_file_path = os.path.join(root, os.path.splitext(os.path.basename(url))[0] + '.pt')
+    if os.path.isfile(cached_vectors_file_path):
+        return(torch.load(cached_vectors_file_path))
+
+    # TODO(Nayef211): determine how to specify a relative file path
```

Comment:
One of the solutions is to hard-code the filename and hashing as a dictionary and save them in `vectors.py` file.

Reply:
Done!